pkg/commands/configstore/describe.go

```diff
@@ -0,0 +1,101 @@
+package configstore
+
+import (
+	"io"
+
+	"github.com/fastly/cli/pkg/cmd"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/cli/pkg/manifest"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/v7/fastly"
+)
+
+// NewDescribeCommand returns a usable command registered under the parent.
+func NewDescribeCommand(parent cmd.Registerer, g *global.Data, m manifest.Data) *DescribeCommand {
+	c := DescribeCommand{
+		Base: cmd.Base{
+			Globals: g,
+		},
+		manifest: m,
+	}
+
+	c.CmdClause = parent.Command("describe", "Retrieve a single config store").Alias("get")
+
+	// Required.
+	c.RegisterFlag(cmd.StoreIDFlag(&c.input.ID)) // --store-id
+
+	// Optional.
+	c.RegisterFlagBool(cmd.BoolFlagOpts{
+		Name:        "metadata",
+		Short:       'm',
+		Description: "Include config store metadata",
+		Dst:         &c.metadata,
+	})
+	c.RegisterFlagBool(c.JSONFlag()) // --json
+
+	return &c
+}
+
+// DescribeCommand calls the Fastly API to describe an appropriate resource.
+type DescribeCommand struct {
+	cmd.Base
+	cmd.JSONOutput
+
+	input    fastly.GetConfigStoreInput
+	manifest manifest.Data
+	metadata bool
+}
+
+// ConfigStoreWithMetadata combines ConfigStore and ConfigStoreMetadata
+// for rendering as JSON and text.
+// The included methods allow for the text package to define a matching
+// interface, which eliminates the circular dependency.
+type ConfigStoreWithMetadata struct {
```

Comment:
Although it does stutter, I'd prefer keeping it named as is for the following:
- This type exists only for the `text.PrintConfigStore` function. That function accepts an [interface](https://github.com/fastly/cli/blob/ad3ce1cf8810c6a064187782befd0a843654d75e/pkg/text/configstore.go#L32-L35) defined by `text` which `ConfigStoreWithMetadata` implements. I think that the interface has a good name, and so this type should be similarly named to make the relationship more clear.
- It's only exported for use by a few test cases. Otherwise, it won't be referenced by other packages (which is where the stutter is noticeable).

Reply:
Thought about this some more, and realized `text.PrintConfigStore` could accept two arguments (instead of the interface) to get the config store and optional metadata. Hopefully this is a less complex approach. bab0af6